sky/backends/cloud_vm_ray_backend.py

```diff
@@ -1194,31 +947,23 @@ def _update_blocklist_on_error(
             assert stderr is None, stderr
             if zones is not None:
                 for zone in zones:
-                    self._blocked_resources.add(
+                    _add_to_blocked_resources(
+                        blocked_resources,
                         launchable_resources.copy(zone=zone.name))
             return False  # definitely_no_nodes_launched
         assert stdout is not None and stderr is not None, (stdout, stderr)
 
         # TODO(zongheng): refactor into Cloud interface?
-        handlers = {
-            clouds.AWS: self._update_blocklist_on_aws_error,
-            clouds.Azure: self._update_blocklist_on_azure_error,
-            clouds.GCP: self._update_blocklist_on_gcp_error,
-            clouds.Lambda: self._update_blocklist_on_lambda_error,
-            clouds.IBM: self._update_blocklist_on_ibm_error,
-            clouds.SCP: self._update_blocklist_on_scp_error,
-            clouds.Local: self._update_blocklist_on_local_error,
-            clouds.Kubernetes: self._update_blocklist_on_kubernetes_error,
-            clouds.OCI: self._update_blocklist_on_oci_error,
-        }
         cloud = launchable_resources.cloud
-        cloud_type = type(cloud)
-        if cloud_type not in handlers:
+        handler = getattr(FailoverCloudErrorHandlerV1,
+                          f'_{str(cloud).lower()}_handler')
+        if handler is None:
             raise NotImplementedError(
                 f'Cloud {cloud} unknown, or has not added '
-                'support for parsing and handling provision failures.')
-        handler = handlers[cloud_type]
-        handler(launchable_resources, region, zones, stdout, stderr)
+                'support for parsing and handling provision failures with '
+                'V1 handlers.')
```

Comment:
```suggestion
                'support for parsing and handling provision failures. '
                'Please implement a handler in FailoverCloudErrorHandlerV2.')
```
..Or should we hint V1 here?

Reply:
Fixed. Thanks!